requre/helpers/files.py

```diff
@@ -102,32 +123,34 @@ def guess_args(cls, func: Callable) -> Any:
 
         @functools.wraps(func)
         def store_files_int(*args, **kwargs):
+            class_test_id_list = [cls.__name__, cls._test_identifier()]
             if not get_if_recording():
                 return func(*args, **kwargs)
             else:
-                current_dir = cls._next_directory_name()
-                os.makedirs(current_dir, exist_ok=True)
                 output = store_function_output(func)(*args, **kwargs)
-                if STORAGE.is_write_mode:
-                    for position in range(len(args)):
-                        arg = args[position]
-                        if not isinstance(arg, str):
-                            continue
-                        if os.path.exists(arg):
-                            current_path = os.path.join(current_dir, str(position))
-                            cls._copy_logic(STORAGE, arg, current_path)
-                    for k, v in kwargs.items():
-                        if os.path.exists(v):
-                            current_path = os.path.join(current_dir, k)
-                            cls._copy_logic(STORAGE, v, current_path)
-                else:
-                    for item in os.listdir(current_dir):
-                        current_path = os.path.join(current_dir, item)
-                        if item.isdigit():
-                            arg = args[int(item)]
+                for position in range(len(args)):
+                    arg = args[position]
+                    if isinstance(arg, str):
+                        keys = class_test_id_list + [position]
+                        if STORAGE.is_write_mode:
+                            if os.path.exists(arg):
+                                cls._copy_logic(STORAGE, pathname=arg, keys=keys)
+                        else:
+                            try:
+                                cls._copy_logic(STORAGE, pathname=arg, keys=keys)
+                            except PersistentStorageException:
+                                pass
```

Comment:
Is it possible to merge this with the second occurrence? (Or simplify that a bit.)

Reply:
hmm, theoretically yes, but not sure if it will be simplier, because, in write mode, exception must not be caught, but raised. so it is question if it will be more readable.

Reply:
done